# Computational approaches for virus-host prediction: A review of methods and applications


This review provides a systematic analysis of the computational phage-host prediction landscape to address current challenges and offer a clear guide for researchers. Our key contributions are:

\begin{itemize}
    \item \textbf{Formalizing the problem:} We delineate the problem formulation of virus-host prediction, creating a clear and structured framework. This allows for a better understanding and comparison of different predictive approaches.
    \item \textbf{Comprehensive survey of biological features:} we present a comprehensive survey of the biological features that have been employed for prediction, detailing the principles behind methods such as direct sequence alignment, CRISPR spacer matching, and alignment-free $k$-mer frequency analysis.
    \item \textbf{Evaluation of existing tools:} we summarize and critically evaluate existing prediction tools, analyzing their respective strengths and weaknesses. This serves as a practical guide for researchers in selecting the appropriate tool for their work.
    \item \textbf{Rigorous and extensive benchmarking:} A key contribution is our rigorous and extensive benchmark of the leading prediction tools. For this, we developed two types of evaluation datasets. \textbf{RefSeq-VHDB: }created using curated phage-host pairs from the RefSeq database. \textbf{MetaHiC:} three independent test sets derived from metagenomic Hi-C (MetaHiC) data to assess performance in more realistic ecological contexts \cite{du2023viralcc, wu2023hi}
\end{itemize}

By synthesizing the current literature and providing a rigorous, comparative performance analysis, this review aims to be a valuable resource that stimulates further innovation in deciphering the complex web of virus-host interactions.